lib/TcpConnection.js

```diff
@@ -21,6 +21,7 @@ const Queue = require('./Queue')
  * @prop {boolean} [recovery] Enable a recovery mode when the TCP connection is lost which store data in a memory queue (FIFO) until the queue max size is reached or the TCP connection is restored. Default: `false`.
  * @prop {({data: Buffer, encoding: string}) => number?} [recoveryQueueSizeCalculation] Function used to calculate the size of stored items. Default: `item => item.data.length + item.encoding.length`.
  * @prop {number?} [recoveryQueueMaxSize] The maximum size of items added to the queue. When reached, oldest items "First In" will be evicted to stay below this size. Default: `1024`.
+ * @prop {(data: Buffer) => Buffer?} [onBeforeDataWrite] Function to manipulate the data before it written.
```

Comment:
This should indicate that the function must be synchronous.

Reply:
Wonder if I should add a callback parameter to allow for async operations.

Reply:
Nevermind, experimented with it, could possibly mess up order of operations. Will keep it sync, and update as such.